packages/kit/src/exports/vite/build/build_server.js

```diff
@@ -101,12 +109,16 @@ export function build_server_nodes(out, kit, manifest_data, server_manifest, cli
 		}
 
 		if (node.universal) {
-			imports.push(
-				`import * as universal from '../${
-					resolve_symlinks(server_manifest, node.universal).chunk.file
-				}';`
-			);
-			exports.push('export { universal };');
+			const page_options = await get_page_options(node);
+			if (!!page_options && page_options.ssr === false) {
+				exports.push(`export const universal = ${s(page_options, null, 2)};`)
+			} else {
+				imports.push(
+					`import * as universal from '../${resolve_symlinks(server_manifest, node.universal).chunk.file}';`
+				);
+				// TODO: when building for analysis, explain why file was loaded on the server if we fail to load it
```

Comment:
TODO do this in analyse.js

```suggestion

```

Reply:
on second thoughts: leave the existing stack trace for now, and finagle this later if we figure out an approach we like